deployment/deploymentClass.js

```diff
@@ -0,0 +1,448 @@
+class Deployment {
+  constructor(depoymentJsonData, args) {
+    // Set arguments to variables
+    this.data = depoymentJsonData;
+    this.access = args.access;
+    this.service = args.service.toLowerCase();
+    this.fork = args.fork.toLowerCase();
+    this.protocol = args.protocol.toLowerCase();
+    this.network = args.network.toLowerCase();
+    this.target = args.target.toLowerCase();
+    this.type = args.type.toLowerCase();
+    this.printlogs = args.printlogs.toLowerCase();
+    this.merge = args.merge.toLowerCase();
+    this.scripts = new Map();
+  }
+
+  prepare() {
+    this.isValidInput();
+    this.flattenFirstJsonLevel();
+    this.isValidJsonData();
+    this.prepareScripts();
+  }
+
+  // Internal functions
+  preprocessing() {
+    this.flattenFirstJsonLevel();
+    this.isValidDeploymentData();
+  }
+
+  // Checks if you are wanting to deploy a single network, all networks for a protocol, or all protocols and networks for a fork
+  prepareScripts() {
+    let scope = this.getDeploymentScope();
+    if (scope == "network") {
+      this.generateScripts(
+        this.protocol,
+        this.network,
+        this.getTemplate(this.protocol, this.network)
+      );
+    } else if (scope == "protocol") {
+      for (const network in this.data[this.protocol]["networks"]) {
+        this.generateScripts(
+          this.protocol,
+          network,
+          this.getTemplate(this.protocol, network)
+        );
+      }
+    } else if (scope == "fork") {
+      let forkProtocols = this.getAllForks();
+      for (const protocol in forkProtocols) {
+        for (const network in this.data[protocol]["networks"]) {
+          this.generateScripts(
+            protocol,
+            network,
+            this.getTemplate(protocol, network)
+          );
+        }
+      }
+    }
+  }
+
+  // Generates scripts necessary for deployment of each network
+  generateScripts(protocol, network, template) {
+    let scripts = [];
+
+    let location = this.getLocation(protocol, network);
+
+    scripts.push("rm -rf build");
+    scripts.push("rm -rf generated");
+    scripts.push("rm -rf results.txt");
+    scripts.push("rm -rf configurations/configure.ts");
+    scripts.push("rm -rf subgraph.yaml");
+
+    scripts.push(
+      "npm run prepare:yaml --PROTOCOL=" +
+        protocol +
+        " --NETWORK=" +
+        network +
+        " --TEMPLATE=" +
+        template
+    );
+    if (
+      this.data[protocol]["networks"][network]["options"][
+        "prepare:constants"
+      ] == true
+    ) {
+      scripts.push(
+        "npm run prepare:constants --PROTOCOL=" +
+          protocol +
+          " --NETWORK=" +
+          network
+      );
+    }
+    scripts.push("graph codegen");
+
+    // We don't want to deploy if we are building or just testing.
+    if (this.type == "deploy" || this.type == "") {
+      scripts.push(this.getDeploymentScript(location));
```

Comment:
okay so looks like this is the case. I would argue you have to explicitly state `deploy` to deploy